org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLSchemaDiagnosticsTest.java

```diff
@@ -481,6 +482,40 @@ public void schema_reference_4_withSchemaLocation() {
 				d(0, 1, 0, 9, XMLSchemaErrorCode.cvc_elt_1_a));
 	}
 
+	@Test
+	public void fuzzyElementNameCodeActionTest() {
+		String xml = 
+		"<project xmlns=\"http://maven.apache.org/POM/4.0.0\" \r\n" +
+		"         xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" +
+		"         xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd\">\r\n" +
+		"    <modules>\r\n" +
+		"      <bodule></bodule>\r\n" + // should be 'module'
+		"    </modules>\r\n" +
+		"</project>";
+		Diagnostic diagnostic = d(4, 7, 4, 13, XMLSchemaErrorCode.cvc_complex_type_2_4_a,
+			"Invalid element name:\n - bodule\n\nOne of the following is expected:\n - module\n\nError indicated by:\n {http://maven.apache.org/POM/4.0.0}\nwith code:");
+		testDiagnosticsFor(xml, diagnostic);
+
+		testCodeActionsFor(xml, diagnostic, ca(diagnostic, te(4, 7, 4, 13, "module"), te(4, 16, 4, 22, "module")));
+	}
+
+	@Test
+	public void fuzzyElementNamesWithOtherOptionsCodeActionTest() {
+		String xml = 
+		"<project xmlns=\"http://maven.apache.org/POM/4.0.0\" \r\n" +
+		"         xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" +
+		"         xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd\">\r\n" +
+		"    <ciManagement>\r\n" +
+		"      <yotifiers></yotifiers>\r\n" + // should be 'notifiers'
+		"    </ciManagement>\r\n" +
+		"</project>";
+		Diagnostic diagnostic = d(4, 7, 4, 16, XMLSchemaErrorCode.cvc_complex_type_2_4_a,
+		"Invalid element name:\n - yotifiers\n\nOne of the following is expected:\n - system\n - url\n - notifiers\n\nError indicated by:\n {http://maven.apache.org/POM/4.0.0}\nwith code:");

```

Comment:
system should be not expected. We should provide similar code actions or other code actions but not the both.